src/telemetry/surveys/SurveyManager.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { callWithTelemetryAndErrorHandling, IActionContext } from 'vscode-azureextensionui';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../localize';
+import nps2 from './nps2';
+
+// Currently-active surveys should be registered here
+const currentSurveys = [nps2];
+
+const surveyRespondedKeyPrefix = 'vscode-docker.surveys.response';
+const surveyFlightPrefix = 'vscode-docker.surveys';
+
+export interface Survey {
+    id: string;
+    url: string;
+    prompt: string;
+    activationDelay: number;

```

Comment:
Consider `activationDelayMs` for extra clarity (but feel free to disregard, I know not everyone likes this)